source/develop/release-management/features/network/provider-physical-network.md

```diff
@@ -0,0 +1,119 @@
+
+## Caveat
+
+The user needs to make sure that the physical network used by the external network is available on the host used by the VM. There is currently now way for Engine to enforce it. The easiest way to accomplish this is to set physical provider network as required for the cluster.
+
+## Implementation
+
+### OpenStack Neutron API
+
+OpenStack Neutron API used by Engine to control external providers already provides means to specify a physical network – attributes `provider:network_type`, `provider:physical_network` and `provider:segmentation_id`. The network types we use are `flat` and `vlan`. Physical network is matched by the VDSM network name. Segmentation ID is used with VLAN networks and matches the VLAN ID defined for the selected network.
+
+### oVirt Provider OVN
+
+The provider reads the attributes described above. If the physical network is defined, the provider implicitly creates a `localnet` port on the given network and attaches it to the received physical network. In case a segmentation ID was passed, it uses it as the VLAN tag on the port.
+
+### VDSM
+
+Localnet port is attached to an abstract network name. This name is mapped to an OVS bridge name on each host via OVS DB attribute `external-ids:ovn-bridge-mappings`. This attribute is configured after each `setupNetworks` command and during upgrades (after reboot or upgrade of vdsm package).
+
+### Engine
+
+The `ProviderNetwork` object in Engine is extended with a link to its physical network (`Network`). In the database, `provider_physical_network_id` is added as a new column to `network` table.
+
+Physical network is used to find the VDSM network name and the VLAN ID. These values are then passed to the network provider.
+
+Engine also covers the validation:
+
+- Physical network must be on the same Data Center as the external network.
+- The custom values (physical network and VLAN ID) must not be specified a when provider physical network is set.
+
+### Engine UI
+
+Even without this feature, it is already possible to set the physical network name and VLAN ID for an external network from `New Network` dialog. Physical network has a separate field, and the VLAN is obtained from the shared `Network Attributes` section. However, this way requires the user to know the VDSM name of the network and manually copy the VLAN ID of the network
+
+For better user experience, this feature exposes an option to select desired provider physical network from Data Center networks to UI.
+
+![add an external network connected to a physical network dialog](/images/features/network/provider-physical-network_new-network-dialog.png)
+
+To expose this feature, an external OVN network is created implicitly when the user adds a new VM network on an OVS cluster. When selecting a network profile for a VM NIC, we list only external networks, not native oVirt networks that triggered their creation.
+
```

Comment:
please add a note about two future additions:
* https://trello.com/c/Uj9xobnx/5-disallow-direct-connection-of-vm-to-a-vm-network-in-ovs-cluster
* https://trello.com/c/hVIVxjB7/24-when-defining-a-physical-network-auto-define-an-external-network-based-on-it